_posts/2020-08-24-gsoc-report-cbouy.md

```diff
@@ -0,0 +1,70 @@
+---
+layout: post
+title: "GSoC Report: From RDKit to the Universe and back 🚀"
+---
+
+With the end of the summer comes the end of my awesome Google Summer of Code adventure with the MDAnalysis team. It's also the occasion for me to report on the code I've implemented, the things I've learned and the challenges I've faced.
+
+## Summary of the project
+---
+
+The goal of my project, "[From RDKit to the Universe and back][project-url]", was to provide interoperability between MDAnalysis and RDKit. *i.e.* to be able to:
+* Read an RDKit molecule and create an MDAnalysis `Universe` with it,
+* Convert an MDAnalysis `AtomGroup` to an RDKit `Molecule`,
+* Leverage RDKit's functionalities directly from MDAnalysis (descriptors, fingerprints, aromaticity perception...*etc.*)
+
+With this in mind, the project was easily cut down in 2 main deliverables (the RDKit Reader/Parser and the Converter) and a few smaller ones for each "wrapper" functionality.
+
+
+## Contributions
+---
+
+### Merged PRs
+
+* **Converting an RDKit molecule to MDAnalysis**: [#2707](https://github.com/MDAnalysis/mdanalysis/pull/2707)  
+  This was the first part of the project and it helped me get acquainted with MDAnalysis as I had never used it before GSoC. The goal here was to be able to "parse" an RDKit molecule and build an MDAnalysis Universe from it.  
+  It was also the occasion for me to start interfacing MDAnalysis with RDKit as I got to implement a new classmethod for the Universe, the `Universe.from_smiles` method which allows us to build a Universe from a SMILES string, but also to add atom selection based on aromaticity.  
+  At the end of this first PR, I was familiar with the different core objects that compose the Universe as well as writing tests with pytest and documentation with sphinx.
+
+* **Converting an MDAnalysis universe to RDKit**: [#2775](https://github.com/MDAnalysis/mdanalysis/pull/2775)  
+  The second goal of the project was to be able to convert an MDAnalysis AtomGroup to an RDKit molecule, allowing users to analyse MD trajectories in RDKit. While this should be trivial once we know how to do the opposite operation, it was actually a real challenge to get a molecule with the correct bond order and charges out of it.  
+  As you may know, most MD topology file formats don't keep track of bond orders and formal charges so we have to find a way to infer this information from what we have. In our case, we require all hydrogen atoms to be explicit in the topology file, as well as elements and bonds (although these two can be guessed). Then the bond orders and formal charges are inferred based on atomic valencies and the number of unpaired electrons, followed by a standardization step of functionnal groups and conjugated systems. This last step is needed because the algorithm implemented to guess bond orders and charges is dependant on the order in which atoms are read.  
```

Comment:
```suggestion
  As you may know, most MD topology file formats don't keep track of bond orders and formal charges so we have to find a way to infer this information from what we have. In our case, we require all hydrogen atoms to be explicit in the topology file, as well as elements and bonds (although these two can be guessed). Then the bond orders and formal charges are inferred based on atomic valencies and the number of unpaired electrons, followed by a standardization step of functional groups and conjugated systems. This last step is needed because the algorithm implemented to guess bond orders and charges is dependent on the order in which atoms are read.  
```